publish.go

```diff
@@ -0,0 +1,326 @@
+	editsExpansionFilesCall := editsExpansionFilesService.Upload(packageName, appEditID, versionCode, expFileType)
+	editsExpansionFilesCall.Media(expansionFile, googleapi.ContentType("application/octet-stream"))
+	if _, err := editsExpansionFilesCall.Do(); err != nil {
+		return fmt.Errorf("failed to upload expansion file, error: %s", err)
+	}
+	log.Infof("Uploaded expansion file %v", expansionFile)
+	return nil
+}
+
+// expFilePth gets the expansion file path from a given config entry
+func expFileInfo(expFileConfigEntry string) (string, string, error) {
+	// "main:/file/path/1.obb"
+	expansionFilePathSplit := strings.Split(expFileConfigEntry, ":")
+	if len(expansionFilePathSplit) < 2 {
+		return "", "", fmt.Errorf("malformed expansion file path: %s", expFileConfigEntry)
+	}
+
+	// "main"
+	expFileType := strings.TrimSpace(expansionFilePathSplit[0])
+	log.Debugf("Expansion file type is %s", expFileType)
+
+	// "/file/path/1.obb"
+	expFilePth := strings.TrimSpace(strings.Join(expansionFilePathSplit[1:], ""))
+	log.Debugf("Expansion file path is %s", expFilePth)
+	return expFilePth, expFileType, nil
+}
+
+// validateExpansionFileConfig validates the given expansion file path. Returns false if it is neither a main or patch
+// file. Example: "main:/file/path/1.obb".
+func validateExpansionFileConfig(expFileEntry string) bool {
+	cleanExpFileConfigEntry := strings.TrimSpace(expFileEntry)
+	return strings.HasPrefix(cleanExpFileConfigEntry, "main:") || strings.HasPrefix(cleanExpFileConfigEntry, "patch:")
+}
+
+// uploadMappingFile uploads the mapping files (that are used for deobfuscation) to Google Play.
+func uploadMappingFile(service *androidpublisher.Service, configs Configs, appEditID string, versionCode int64) error {
+	log.Debugf("Getting mapping file from %v", configs.MappingFile)
+	mappingFile, err := os.Open(configs.MappingFile)
+	if err != nil {
+		return fmt.Errorf("failed to read mapping file (%s), error: %s", configs.MappingFile, err)
+	}
+	log.Debugf("Uploading mapping file %v with package name '%v', AppEditId '%v', version code '%v'", configs.MappingFile, configs.PackageName, appEditID, versionCode)
+	editsDeobfuscationFilesService := androidpublisher.NewEditsDeobfuscationfilesService(service)
+	editsDeobfuscationFilesUploadCall := editsDeobfuscationFilesService.Upload(configs.PackageName, appEditID, versionCode, "proguard")
+	editsDeobfuscationFilesUploadCall.Media(mappingFile, googleapi.ContentType("application/octet-stream"))
+
+	if _, err = editsDeobfuscationFilesUploadCall.Do(); err != nil {
+		return fmt.Errorf("failed to upload mapping file, error: %s", err)
+	}
+
+	log.Printf(" uploaded mapping file for apk version: %d", versionCode)
+	return nil
+}
+
+// expansionFiles gets the expansion files from the received configuration. Returns true and the entries (type and
+// path) of them when any found, false or error otherwise.
+func expansionFiles(appPaths []string, expansionFilePathConfig string) (bool, []string, error) {
```

Comment:
Bool retrun type is not really needed, if returned array is empty can mean the same thing.
Would move this function to config.go, is more related.

Reply:
Removed bool return type.

Reply:
Moved to config.